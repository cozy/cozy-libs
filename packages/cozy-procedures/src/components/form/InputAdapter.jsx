import React from 'react'
import PropTypes from 'prop-types'
import { Input, translate } from 'cozy-ui/transpiled/react'

import stripInvalidInputProps from './stripInvalidInputProps'

const InputAdapter = props => (
  <Input
    {...stripInvalidInputProps(props)}
    onChange={e => {
      props.onChange(e.target.value)
    }}
  />
)

InputAdapter.propTypes = {
  onChange: PropTypes.func.isRequired
}

export default translate()(InputAdapter)
