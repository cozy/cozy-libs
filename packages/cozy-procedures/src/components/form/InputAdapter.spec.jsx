import React from 'react'

import tMock from '../../../test/__mocks__/tMock'
import { InputAdapter } from './InputAdapter'

describe('InputAdapter component', () => {
  it('should match snapshot', () => {
    const onChangeSpy = jest.fn()
    const props = {
      onChange: onChangeSpy,
      t: tMock,
      schema: {
        type: 'string'
      }
    }
    const component = shallow(<InputAdapter {...props} />)
    expect(component).toMatchSnapshot()
  })
})
