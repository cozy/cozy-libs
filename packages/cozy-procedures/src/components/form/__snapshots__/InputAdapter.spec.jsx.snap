// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAdapter component should match snapshot 1`] = `
<Input
  error={false}
  fullwidth={false}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  size="large"
  type="text"
/>
`;
